tests/foreman/ui/test_lifecycleenvironment.py

```diff
@@ -336,3 +351,61 @@ def test_positive_custom_user_view_lce(self):
                 session.nav.go_to_users()
             # assert that the user can view the lvce created by admin user
             self.assertIsNotNone(self.lifecycleenvironment.search(env_name))
+
+    @run_only_on('sat')
+    @tier3
+    def test_positive_search_lce_content_view_packages(self):
+        """Search Lifecycle Environment content view packages
+
+        :id: bceb97ac-cd20-4a45-a250-f26c194f651a
+
+        :steps:
+            1. Create a product with a repository synchronized
+                - The repository must contain at least two package names P1 and
+                  P2
+                - P1 has only one package version
+                - P2 has two package versions
+            2. Create a content view with the repository and publish it
+            3. Go to Lifecycle Environment > Library > Packages
+            4. Select the content view
+            5. Search by exact package names with versions
+            6. Search by package names
+
+        :expectedresults: only the searched packages where found
+
+        :BZ: 1432155
+
+        :CaseLevel: System
+        """
+        packages = [
+            {'name': FAKE_0_CUSTOM_PACKAGE_NAME,
+             'versions': [FAKE_0_CUSTOM_PACKAGE]},
+            {'name': FAKE_1_CUSTOM_PACKAGE_NAME,
+             'versions': [FAKE_1_CUSTOM_PACKAGE, FAKE_2_CUSTOM_PACKAGE]},
+        ]
+        org = entities.Organization().create()
+        product = entities.Product(organization=org).create()
+        repository = entities.Repository(
+            product=product, url=FAKE_0_YUM_REPO).create()
+        repository.sync()
+        content_view = entities.ContentView(
+            organization=org, repository=[repository]).create()
+        content_view.publish()
+        with Session(self) as session:
+            set_context(session, org=org.name)
+            for package in packages:
+                for package_version in package['versions']:
+                    names_found = self.lifecycleenvironment.get_yum_packages(
+                        ENVIRONMENT,
+                        package_version,
+                        cv_name=content_view.name
+                    )
+                    self.assertEqual(len(names_found), 1)
+                    self.assertEqual(names_found[0], package['name'])
+            for package in packages:
```

Comment:
redundant line of code

Reply:
where is the redundant and do not get your idea, we have many packages to search for

Reply:
the first loop search by full package name with versions
the second search by package name only

Reply:
it will do the same without second `for package in packages:`

Reply:
yes we can